src/views/catalog/utils/WizardTemplateContext.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+
+type WizardTemplateContextType = {
+  /** the template used for the wizard */
+  template?: V1Template;
+  /**
+   * updates the template. The provided object will be merged with the existing template object
+   * @param template V1Template
+   */
+  updateTemplate?: (template: V1Template) => void;
+};
+
+export const WizardTemplateContext = React.createContext<WizardTemplateContextType>({});
+
+export const WizardTemplateContextProvider = WizardTemplateContext.Provider;
+
+export const useWizardTemplate = (): WizardTemplateContextType => {
+  const [template, setTemplate] = React.useState<V1Template>(() => getSessionStorage('template'));
+
+  React.useEffect(() => {
+    setSessionStorage('template', template);
+  }, [template]);
+
+  return {
+    template,
+    updateTemplate: (updatedTemplate: V1Template) =>
+      setTemplate((prevTemplate) => ({
+        ...prevTemplate,
+        ...updatedTemplate,
+      })),
+  };
+};
+
+function setSessionStorage(key: string, value) {
+  window.sessionStorage.setItem(key, JSON.stringify(value));
+}
+
+function getSessionStorage(key: string) {
+  const value = window.sessionStorage.getItem(key);
+  return value ? JSON.parse(value) : undefined;
```

Comment:
JSON is not JavaScript. It uses a limited subset of the JavaScript syntax, but the only data types it can encode are:

null
boolean
number
string
object: map from string to any of the types
array: list of values
(see http://json.org/)

JSON is not capable of serializing functions. Any functions will be omitted (like undefined), prototypes will be destroyed, etc.

If we going save or pass any function we should avoid it